extension/show-names.js

```diff
@@ -0,0 +1,56 @@
+window.showRealNames = () => {
+	const storageKey = 'cachedNames';
+
+	const getCachedUsers = cb => {
+		chrome.storage.local.get(storageKey, data => cb(data[storageKey]));
+	};
+
+	const updateCachedUsers = users => {
+		chrome.storage.local.set({[storageKey]: users});
+	};
+
+	const addUsersName = (user, name) => {
+		const $usernameLinks = $(`.timeline-comment-header-text:not(.has-full-name) a[href="/${user}"]`);
+		$usernameLinks.each((i, userLink) => {
+			$(`<span class="comment-full-name">${name} -</span>`).insertAfter(userLink);
+			$(userLink).closest('.timeline-comment-header-text').addClass('has-full-name');
+		});
+	};
+
+	getCachedUsers((users = {}) => {
+		const usersOnPage = $('.js-discussion .author').dom.map(el => el.innerText);
+		const uniqueUsers = new Set(usersOnPage);
+
+		// Add cached users to DOM first, since the calls for everyone else will be slow
+		for (const user of uniqueUsers) {
+			const cachedName = users[user];
+			if (cachedName) {
+				addUsersName(user, cachedName);
+				uniqueUsers.delete(user);
+			}
+		}
+
+		const userUrl = user => `https://github.com/${user}/following`;
+		const requests = Array.from(uniqueUsers).map(user => fetch(userUrl(user)).then(res => res.text()));
+		Promise.all(requests).then(profiles => {
+			const userCache = {};
+
+			for (const profile of profiles) {
+				// `createContextualFragment` because Sprint/jQuery $ fn treats an entire HTML document as one giant text node
+				const frag = document.createRange().createContextualFragment(profile);
+				const fullnameEl = frag.querySelector('h1 strong');
+				const usernameEl = frag.querySelector('h1 a:nth-of-type(2)');
+				const fullname = fullnameEl && fullnameEl.innerText;
+				const username = usernameEl && usernameEl.innerText;
```

Comment:
You already have the username on line 34. I would rather just pass that around that fetching it again.